persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/TreatmentMapper.java

```diff
@@ -7,13 +7,13 @@
 import java.util.Set;
 
 public interface TreatmentMapper {
-    List<Treatment> getAllTreatments(List<String> sampleIds, List<String> studyIds);
+    List<Treatment> getAllTreatments(List<String> sampleIds, List<String> studyIds, List<String> uniqueStudyIds);
 
-    List<ClinicalEventSample> getAllSamples(List<String> sampleIds, List<String> studyIds);
+    List<ClinicalEventSample> getAllSamples(List<String> sampleIds, List<String> studyIds, List<String> uniqueStudyIds);
 
-    Set<String> getAllUniqueTreatments(List<String> sampleIds, List<String> studyIds);
+    Set<String> getAllUniqueTreatments(List<String> sampleIds, List<String> studyIds, List<String> uniqueStudyIds);
```

Comment:
might be cleaner if the unique/distinct step can be done in mysql in some way. Looks slightly odd when studyIds are given as argument twice imo